### ISSUE TITLE
rc1-1.2.1 -- possible random client crashes?

### ISSUE THREAD

**Issue by brahaney** [Mydayyy/Valheim-ServerSideMap#19]
Since updating, myself and one of my players have experienced random client freeze/crash to desktop. There are no log entries in the player log or server log to indicate what happened. Upon re-joining the game, inventory and player location is reset to what it was at previous login. The only change to our server/client environment is this mod update. Is there a way to get debug logging from the mod, or all the mods?

The crashes are infrequent (twice in 6 hours for my friend, twice in about 6 hours for me) so I've disabled the mod again to see if I continue to observe the crashes. But with no evidence of an issue in the logs I'm not sure how I could correlate the crash to anything in particular. (we also enabled pin sharing last night. Unsure if that could have anything to do with it.)

**Comment by Mydayyy** [comment]
Greetings,

I didnt really have any long playtest sessions yet with the new version, but there arent really any changes which are executed clientside compared to the previous 1.2.0 release. All new code is only executed on the server.

Do users of your users not have this problem while running the mod? Is everyone running into this issue?

Cheers
Mydayyy

**Comment by brahaney** [comment]
only two of us have the mod installed right now. the other two players are vanilla clients. and myself and the other player with the mod had like a 10 hour marathon session the previous night without issue with SSM 1.2.0. Since updating last night we've had 4 crashes between the two of us in about 12 hours combined playtime. I don't know that it IS this mod, I just am unsure what else it might be.

**Comment by Mydayyy** [comment]
What you could try is to revert to 1.2.0 and see if the problem occurs again. 1.2.0 is compatible with a serverside 1.2.1 version.

If you really want to you could also try to run one person on 1.2.0 and one person 1.2.1 and kind of do an A/B test. But I wouldn't have an idea where the issue should be, because as mentioned, the client does absolutely not have any new code that is used/executed

**Comment by brahaney** [comment]
I'll downgrade my client before the next time I play and see if we get crashes with 1.2.0 client/1.2.1 server. I'll continue to have a couple of vanilla client players too.

**Comment by brahaney** [comment]
ok. in 2 hours' time, just letting the game idle in the background while I work, it's crashed twice. This is with 1.2.0 client/1.2.1 server. after the first crash I removed ValheimPlus and CraftFromContainers mods, so that SSM was the only mod present, and then got a 2nd crash about an hour later.

**Comment by Mydayyy** [comment]
Keep me posted!

Here's me lowkey hoping it doesn't turn out to be ssm, because I'd have literally zero clue where the issue could be. We didnt experience any crashes so far on my server, but we only run SSM as the sole mod. So far I didnt receive any other reports about crashes too.

Cheers
Mydayyy

**Comment by brahaney** [comment]
sorry for the long silence-- been busy with work. But after some long testing, I've come to the conclusion that I have no idea what was going on. The crashes were consistent for two days, while I enabled/disabled every combination of mods I could think of, and even with no mods enabled. I eventually rebuilt my server from scratch with all the same mods and versions and we haven't had any crashes since. I'm closing this out. Cannot reproduce.
